src/layouts/sidebar-sidecar/components/table-of-contents/use-active-section.ts

```diff
@@ -138,14 +154,7 @@ export function useActiveSection(
     })
 
     return () => {
-      headings.forEach((section) => {
-        const el = document
-          .getElementById('main')
-          ?.querySelector(`#${section.slug}`)
-        if (el) {
-          observer.unobserve(el)
-        }
-      })
```

Comment:
This was a funky one, because we were iterating through each heading and unobserving like this, on navigation from a page with a sidecar to another page with a sidecar, an old observer was still firing if there were headings with the same slug on the new page. Switched to `.disconnect()` here to avoid that.